x/merkledb/intermediate_node_db.go

```diff
@@ -27,7 +29,9 @@ type intermediateNodeDB struct {
 	// from the cache, which will call [OnEviction].
 	// A non-nil error returned from Put is considered fatal.
 	// Keys in [nodeCache] aren't prefixed with [intermediateNodePrefix].
-	nodeCache onEvictCache[Key, *node]
+	writeBuffer onEvictCache[Key, *node]
+
+	nodeCache cache.Cacher[Key, *node]
```

Comment:
If the comment above `writeBuffer` also applies to `nodeCache` I think you should remove the `\n` above.

Reply:
nodecache wont cause writes on evictions, but the thing about nils is still true, may just replicate the parts that apply.